### PR TITLE
Inline the definitions of some utility functions.

### PR DESCRIPTION
**Pull request by robdockins**
This significantly speeds up some workloads.

### CODE REVIEW

File: what4/src/What4/Protocol/SMTWriter.hs [hunk 1]
```diff
@@ -787,6 +786,23 @@ cacheLookup
 cacheLookup conn lookup_action =
   readIORef (entryStack conn) >>= firstJustM lookup_action
 
+
+-- | Like 'findM', but also allows you to compute some additional information in the predicate.
+firstJustM :: Monad m => (a -> m (Maybe b)) -> [a] -> m (Maybe b)
+firstJustM _ [] = pure Nothing
+firstJustM p (x:xs) = maybeM (firstJustM p xs) (pure . Just) (p x)
```
Comment (kquick) [inline what4/src/What4/Protocol/SMTWriter.hs:793]: I normally use `return` in the `Monad` context and only use `pure` in `Applicative` contexts.

**Review by kquick — APPROVED** [review]
Minor suggestion above, but looking forward to enjoying this speedup!

**Comment by robdockins** [comment]
Do we expect that the 9.0.1 builds are failing?

**Comment by RyanGlScott** [comment]
I certainly wouldn't have expected them to fail. For reference, both of the test failures look like [this](https://github.com/GaloisInc/what4/pull/141/checks?check_run_id=3204470610#step:12:112):

```
    boolector:                                                                  FAIL
      Exception: user error (Parse exception: Failed reading: empty)
```

Comparing the CI logs from a GHC 8.10.4 job versus a 9.0.1 one, the 8.10.4 job [used Boolector 3.2.1](https://github.com/GaloisInc/what4/pull/141/checks?check_run_id=3204470635#step:12:18), while the 9.0.1 job [used Boolector 3.2.2](https://github.com/GaloisInc/what4/pull/141/checks?check_run_id=3204470610#step:12:18). Moreover, the most recent 9.0.1 CI run on `master` (from 24 days ago) [used Boolector 3.2.1](https://github.com/GaloisInc/what4/runs/3003764104#step:12:18). Perhaps this version difference is to blame?

**Comment by robdockins** [comment]
Seems likely.  I'll try to reproduce locally and open a ticket.

**Pull request merged**